src/main/java/org/example/homework4/Task1.java

```diff
@@ -0,0 +1,22 @@
+package org.example.homework4;
+
+public class Task1 {
+    public static void main(String[] args) {
+        int year = 1889;
+        boolean leapYear = isLeapYear(year);
+        System.out.println("Year " + year + " is" + (leapYear ? "" : " not") + " leap");
+    }
+
+    private static boolean isLeapYear(int year) {
+        if (year < 1 || year > 9999) {
+            return false;
+        }
+        if (year % 4 == 0 && year % 100 != 0) {
```

Comment:
This is correct, but I personally prefer doing smth like this
```
        if ((year % 4 == 0 && year % 100 != 0) || (year % 400 == 0)) {
            return true;
        }
 ```
 In your case you have 2 separate expressions where return result is identical(true), but they could be combined into one

Reply:
Done)